batchHeader.go

```diff
@@ -213,6 +211,30 @@ func (bh *BatchHeader) String() string {
 	return buf.String()
 }
 
+// Equal returns true only if two BatchHeaders are equal.
+// Equality is determined by the Nacha defined fields of each record.
+func (bh *BatchHeader) Equal(other *BatchHeader) bool {
+	if bh.ServiceClassCode != other.ServiceClassCode {
+		return false
+	}
+	if !strings.EqualFold(bh.CompanyName, other.CompanyName) {
+		return false
+	}
+	if bh.CompanyIdentification != other.CompanyIdentification {
+		return false
+	}
+	if bh.StandardEntryClassCode != other.StandardEntryClassCode {
+		return false
+	}
+	if bh.EffectiveEntryDate != other.EffectiveEntryDate {
+		return false
+	}
+	if bh.ODFIIdentification != other.ODFIIdentification {
+		return false
+	}
```

Comment:
```suggestion
	if bh.ServiceClassCode != other.ServiceClassCode &&
	!strings.EqualFold(bh.CompanyName, other.CompanyName) &&
	bh.CompanyIdentification != other.CompanyIdentification &&
	bh.StandardEntryClassCode != other.StandardEntryClassCode &&
        bh.EffectiveEntryDate != other.EffectiveEntryDate &&
	bh.ODFIIdentification != other.ODFIIdentification {
		return false
	}
```

Reply:
I was debating this. Putting it all into one `if` seems a bit messy. We don't really do that elsewhere in this project.

Reply:
I'm fine with leaving as is